firedrake/output.py

```diff
@@ -464,6 +470,39 @@ def _prepare_output(self, function, max_elem):
 
         return OFunction(array=get_array(output), name=name, function=output)
 
+    def _prepare_coordinates(self, mesh, max_elem):
+        from firedrake import Function, Interpolator
+        from tsfc.finatinterface import create_element as create_finat_element
+
+        function = mesh.coordinates
+        name = function.name()
+        # Need to project/interpolate?
+        # If space is not the max element, we must do so.
+        finat_elem = function.function_space().finat_element
+        if finat_elem == create_finat_element(max_elem):
+            return OFunction(array=get_array(function),
+                             name=name, function=function)
+        #  OK, let's go and do it.
+        output = self._output_functions.get(function)
+        if output is None:
+            V = self._get_output_functionspace(function, max_elem)
+            ufl_elem = V.ufl_element().sub_elements()[0]
+            interpolator = mesh.aux_coord_interpolators.get(ufl_elem)
+            if interpolator is None:
+                # allocate a new coordinate function
+                print(f'Allocating coords for V={V}')
+                output = Function(V)
+                interpolator = Interpolator(function, output)
+                # store weak reference in the mesh
+                # need to store the interpolator object too
+                mesh.aux_coord_interpolators[ufl_elem] = interpolator
+                interpolator.interpolate()
+            output = interpolator.V  # HACK I know that V is the output function, not the space ...
```

Comment:
I think this is currently the only way to get the target from `Interpolator`. 
Would be cleaner if the target space and function were separate properties in `Interpolator`.